tilavarauspalvelu/integrations/email/template_context/reservation.py

```diff
@@ -0,0 +1,681 @@
+            "non_subsidised_price": reservation.non_subsidised_price,
+            "confirmed_instructions": reservation.actions.get_instructions(kind="confirmed", language=language),
+            **params_for_base_info(reservation=reservation, language=language),
+            **params_for_price_info(reservation=reservation, language=language),
+        }
+
+    text_reservation_approved = (
+        pgettext("Email", "Your booking has been confirmed with the following discount:")
+        if data["price"] < data["non_subsidised_price"]
+        else pgettext("Email", "Your booking is now confirmed")
+    )
+
+    return {
+        "title": pgettext("Email", "Your booking is confirmed"),
+        "text_reservation_approved": text_reservation_approved,
+        "instructions_label": pgettext("Email", "Additional information about your booking"),
+        "instructions": data["confirmed_instructions"],
+        **get_contex_for_base_template(email_recipient_name=data["email_recipient_name"], language=language),
+        **get_contex_for_reservation_basic_info(
+            reservation_unit_name=data["reservation_unit_name"],
+            unit_name=data["unit_name"],
+            unit_location=data["unit_location"],
+            begin_datetime=data["begin_datetime"],
+            end_datetime=data["end_datetime"],
+            language=language,
+        ),
+        **get_contex_for_reservation_price(
+            price=data["price"],
+            tax_percentage=data["tax_percentage"],
+            booking_number=data["booking_number"],
+            language=language,
+        ),
+        **get_contex_for_reservation_manage_link(language=language),
+        **get_contex_for_closing_polite(language=language),
+    }
+
+
+@overload
+def get_context_for_reservation_requires_handling(
+    reservation: Reservation,
+    *,
+    language: Lang,
+) -> EmailContext: ...
+
+
+@overload
+def get_context_for_reservation_requires_handling(
+    *,
+    language: Lang,
+    email_recipient_name: str,
+    reservation_unit_name: str,
+    unit_name: str,
+    unit_location: str,
+    begin_datetime: datetime.datetime,
+    end_datetime: datetime.datetime,
+    price: Decimal,
+    subsidised_price: Decimal,
+    tax_percentage: Decimal,
+    booking_number: int,
+    pending_instructions: str,
+) -> EmailContext: ...
+
+
+@get_translated
+def get_context_for_reservation_requires_handling(
+    reservation: Reservation | None = None,
+    *,
+    language: Lang,
+    **data: Any,
+) -> EmailContext:
+    if reservation is not None:
+        begin_datetime = reservation.begin.astimezone(DEFAULT_TIMEZONE)
+        end_datetime = reservation.end.astimezone(DEFAULT_TIMEZONE)
+        data: dict[str, Any] = {
+            "email_recipient_name": reservation.actions.get_email_reservee_name(),
+            "subsidised_price": reservation.actions.calculate_full_price(begin_datetime, end_datetime, subsidised=True),
```

Comment:
The old version works (a bit obscurely) so that `subsidised_price` will be different to `reservation.price` only if discount has been applied for: https://github.com/City-of-Helsinki/tilavarauspalvelu-core/blob/175161b2419810643679c430dd99b6f6f961f0c9/tilavarauspalvelu/utils/email/email_builder_reservation.py#L209
That is because if it is has _not_ been applied for, then the final price _should not_ be changed from the default highest price in the handling (even if it theoretically could be changed by a handler).

The condition (`if subsidised_price < price`) in the `reservation_price_range` template partials has the benefit, though, that if there is no range defined in the reservation unit's pricing (but the form has allowed applying for a discount and the user has), then it won't print out a bogus price range such as 15,00 - 15,00 € -- unlikely as that combination of data perhaps is.

Even so, maybe the clearest solution (in this specific type of email context) would be to always calculate `subsidised_price` but in the `reservation_price_range` templates rewrite the condition as e.g. `if price_can_be_subsidised`, while price_can_be_subsidised would be here set to true if `reservation.applying_for_free_of_charge & subsidised_price <price`.

Reply:
Modified the logic to suit this and added tests